src/pathfora.js

```diff
@@ -398,17 +399,29 @@
      * @param {obj} widget
      * @throws {Error} error
      */
-    insertWidget: function (segment, widget, config) {
+    insertWidget: function (method, segment, widget, config) {
       // assume that we need to add a new widget until proved otherwise
       var makeNew = true;
+      var subject;
 
       // make sure our scaffold is valid
       if(!config.target){
         throw new Error('Invalid scaffold. No target array.');
       }
+      if(!config.exclude){
+        throw new Error('Invalid scaffold. No exclude array.');
+      }
+
+      if (method === "target"){
+        subject = config.target;
+      }else if(method === "exclude"){
+        subject = config.exclude;
+      }else{
```

Comment:
yah i wasnt even sure what common was doing so didnt include it in the scaffold.